### PR TITLE
Updates to accommodate API changes in lib_xcore_math v2.1.0

### PR DESCRIPTION
**Pull request by keithm-xmos**
#### Includes

The recommended API include is now:

`
#include "xmath/xmath.h"
`
All prior lib_xs3_math v1 header includes were changed.

#### Renamed functions

All occurrences of `double_to_float_s32` were replaced with `f64_to_float_s32`
All occurrences of `float_to_float_s32` were replaced with `f32_to_float_s32`
The `xs3_` prefix was removed from all function calls to the older, lib_xs3_math API

#### Typedef changes

The type `fixed_s32_t` has been removed from the library.  

`fixed_s32_t` types were replaced with `int32_t` where the q-format was uncertain.  
`fixed_s32_t` types  were replaced with the new typedefs `uq0_32`, `uq1_31`, `uq2_30`, `uq8_24` (or appropriate) where the q-format could be determined from the variable name.  

### Notes to Reviewers

- Ensure the python that generates source code is correct

- Ensure that the q-format types are correct

The remainder of the changes should be straight forward.  Mostly  needs another set of eyes to make sure no silly mistakes were made.  120 files have been modified.

### CODE REVIEW

File: modules/lib_vnr/api/features/vnr_features_state.h [hunk 1]
```diff
@@ -64,7 +65,7 @@ typedef struct {
  */
 typedef struct {
     /** Feature buffer containing the most recent VNR_MEL_FILTERS frames' MEL frequency spectrum. */
-    fixed_s32_t DWORD_ALIGNED feature_buffers[VNR_PATCH_WIDTH][VNR_MEL_FILTERS];
+    int32_t DWORD_ALIGNED feature_buffers[VNR_PATCH_WIDTH][VNR_MEL_FILTERS];
```
Comment (uvvpavel) [inline modules/lib_vnr/api/features/vnr_features_state.h:68]: This should be q24

File: modules/lib_vnr/src/features/vnr_features_priv.h [hunk 2]
```diff
@@ -46,7 +46,7 @@ void vnr_priv_make_slice(fixed_s32_t *new_slice, const bfp_complex_s32_t *X, int
  *
  * This function name matches with the corresponding function in py_vnr python model.
  */
-void vnr_priv_add_new_slice(fixed_s32_t (*feature_buffers)[VNR_MEL_FILTERS], const fixed_s32_t *new_slice);
+void vnr_priv_add_new_slice(int32_t (*feature_buffers)[VNR_MEL_FILTERS], const uq8_24 *new_slice);
```
Comment (uvvpavel) [inline modules/lib_vnr/src/features/vnr_features_priv.h:49]: feature_buffers should be q24

File: examples/bare-metal/shared_src/external_deps/fetch_deps.cmake [hunk 3]
```diff
@@ -2,13 +2,13 @@
 include(FetchContent)
 
 FetchContent_Declare(
-    xs3_math
-    GIT_REPOSITORY https://github.com/xmos/lib_xs3_math.git
-    GIT_TAG        881cf848a5899f4ecb9182e7f50fd3705e5b68ef
+    xcore_math
+    GIT_REPOSITORY https://github.com/xmos/lib_xcore_math.git
+    GIT_TAG        v2.1.0
```
Comment (uvvpavel) [inline examples/bare-metal/shared_src/external_deps/fetch_deps.cmake:7]: I think v2.1.1 is already available. Can we use the newer one?

File: test/lib_vnr/vnr_unit_tests/feature_extraction/test_vnr_priv_make_slice.py [hunk 4]
```diff
@@ -15,7 +15,7 @@ def test_vnr_priv_make_slice(target, tflite_model):
 
     input_data = np.empty(0, dtype=np.int32)
     input_words_per_frame = fp.FRAME_ADVANCE + 1 #No. of int32 values sent to dut as input per frame
-    output_words_per_frame = fp.MEL_FILTERS # MEL_FILTERS fixed_s32_t values. Exponent fixed to -24
+    output_words_per_frame = fp.MEL_FILTERS # MEL_FILTERS int32_t values. Exponent fixed to -24
```
Comment (uvvpavel) [inline test/lib_vnr/vnr_unit_tests/feature_extraction/test_vnr_priv_make_slice.py:18]: This should be q24

**Review by uvvpavel — CHANGES_REQUESTED** [review]
Thanks for making this PR @keithm-xmos. Few undefined fixed point types, otherwise looks good!

**Review by uvvpavel — APPROVED** [review]
Looks good

**Review by astewart-xmos — APPROVED** [review]

**Pull request merged**